cfgov/paying_for_college/views.py

```diff
@@ -0,0 +1,339 @@
+import json
+import os
+import re
+from collections import OrderedDict
+
+from django.core.urlresolvers import reverse
+from django.http import HttpResponse, HttpResponseBadRequest
+from django.shortcuts import get_object_or_404, render
+from django.utils import timezone
+from django.views.generic import TemplateView, View
+from haystack.query import SearchQuerySet
+
+from paying_for_college.disclosures.scripts import nat_stats
+from paying_for_college.forms import FeedbackForm
+from paying_for_college.models import (
+    ConstantCap, ConstantRate, Feedback, Notification, Program, School
+)
+
+
+BASEDIR = os.path.dirname(__file__)
+DISCLOSURE_ROOT = 'paying-for-college2'
+EXPENSE_FILE = '{}/fixtures/bls_data.json'.format(BASEDIR)
+IPED_ERROR = "noSchool"
+OID_ERROR = "noOffer"
+PID_ERROR = "noProgram"
+
+
+def get_json_file(filename):
+    try:
+        with open(filename, 'r') as f:
+            return f.read()
+    except Exception:
+        return ''
+
+
+def validate_oid(oid):
+    """
+    make sure an oid contains only hex values 0-9 a-f A-F and is 40 characters
+    return True if the oid is valid
+    """
+    find_illegal = re.search('[^0-9a-fA-F]+', oid)
+    if find_illegal:
+        return False
+    else:
+        if len(oid) >= 40 and len(oid) <= 128:
+            return True
+        else:
+            return False
+
+
+def validate_pid(pid):
+    if not pid:
+        return False
+    for char in [';', '<', '>', '{', '}']:
+        if char in pid:
+            return False
+    return True
+
+
+def get_program_length(program, school):
+    if program and program.level:
+        LEVEL = program.level
+    elif school and school.degrees_predominant:
+        LEVEL = school.degrees_predominant
+    elif school and school.degrees_highest:
+        LEVEL = school.degrees_highest
+    else:
+        return None
+    if LEVEL in ['0', '1', '2']:
+        return 2
+    elif LEVEL in ['3', '4']:
+        return 4
+    else:
+        return None
+
+
+def get_school(schoolID):
+    """Try to get a school by ID; return either school or empty string"""
+    try:
+        school = School.objects.get(school_id=int(schoolID))
+    except Exception:
+        return None
+    else:
+        if school.operating is False:
+            return None
+        else:
+            return school
+
+
+def get_program(school, programCode):
+    """Try to get latest program; return either program or empty string"""
+    if not validate_pid(programCode):
+        return None
+    programs = Program.objects.filter(program_code=programCode,
+                                      institution=school).order_by('-pk')
+    if programs:
+        return programs[0]
+    else:
+        return None
+
+
+class BaseTemplateView(TemplateView):
+
+    def get_context_data(self, **kwargs):
+        context = super(BaseTemplateView, self).get_context_data(**kwargs)
+        context['base_template'] = 'front/base_update.html'
```

Comment:
I imagine that this could maybe be cleaned up further in future since the base template and root are fixed now that there's no standalone mode.

Reply:
Yes indeed.